### PR TITLE
Argument list for "python -m virtualenv" without empty strings

### LINKED ISSUE #7322: Empty arguments not supported w/ python --mvirtualenv /dest/
## Details

Firstly, I have a patch that's tested and works and will submit a PR soon :innocent: 

In relation a set of specific deployment needs, I found it necessary to try out Read The Docs on a newer platform (Python 3.7), and I think it causes the following breakage:

image

* Read the Docs project URL: n/a
* Build URL (if applicable): n/a
* Read the Docs username (if applicable): n/a

The issue is also reported 14 days ago here:

https://stackoverflow.com/questions/62050977/readthedocs-virtualenv-error-unrecognized-arguments-during-update-repos

It was also reported in #6915 but closed for the wrong reasons (it's not due to an environment variable or similar)

## Reproducing

If there is no option `--system-site-packages` passed to `python3 -mvirtualenv`, it will be invoked like this, resulting in an error:

```
(venv) readthedocs@6d51bf78d765:~/git$ python3.7 -mvirtualenv "" /usr/share/readthedocs/git/user_builds/myproject/envs/latest
usage: virtualenv [--version] [--with-traceback] [-v | -q] [--app-data APP_DATA] [--clear-app-data] [--discovery {builtin}] [-p py] [--creator {builtin,cpython3-posix,venv}] [--seeder {app-data,pip}] [--no-seed]
                  [--activators comma_sep_list] [--clear] [--system-site-packages] [--symlinks | --copies] [--no-download | --download] [--extra-search-dir d [d ...]] [--pip version] [--setuptools version] [--wheel version] [--no-pip]
                  [--no-setuptools] [--no-wheel] [--symlink-app-data] [--prompt prompt] [-h]
                  dest
virtualenv: error: unrecognized arguments: /usr/share/readthedocs/git/user_builds/myproject/envs/latest
```

## Expected Result

I'm pretty sure this is not reproducible for the core team's Docker containers. But you'll probably see it once the Python environments are updated from Python 3.6 to Python 3.7.

## Actual Result

My environment fails executing because of an empty-string command line argument.

### PR DESCRIPTION
**Pull request by benjaoming**
Fixes #7322

### CODE REVIEW

File: readthedocs/doc_builder/python_environments.py [hunk 1]
```diff
@@ -295,19 +295,27 @@ def venv_path(self):
...
     def setup_base(self):
-        site_packages = ''
+        """Create a virtualenv, invoking python -m virtualenv"""
+
+        # --no-download was removed because of the pip breakage,
+        # it was sometimes installing pip 20.0 which broke everything
+        # https://github.com/readthedocs/readthedocs.org/issues/6585
+
+        # Important not to add empty string arguments, see:
+        # https://github.com/readthedocs/readthedocs.org/issues/7322
```
Comment (stsewd) [inline readthedocs/doc_builder/python_environments.py:305]: We can move these two notes into the docstring with `.. note::`

Comment (benjaoming) [reply readthedocs/doc_builder/python_environments.py]: Cool!

File: readthedocs/doc_builder/python_environments.py [hunk 2]
```diff
@@ -295,19 +295,31 @@ def venv_path(self):
         return os.path.join(self.project.doc_path, 'envs', self.version.slug)
 
     def setup_base(self):
-        site_packages = ''
+        """
+        Create a virtualenv, invoking python -m virtualenv
```
Comment (stsewd) [inline readthedocs/doc_builder/python_environments.py:299]:
```suggestion
        Create a virtualenv, invoking python -m virtualenv.
```

The linter will fail bc of this

Comment (benjaoming) [reply readthedocs/doc_builder/python_environments.py]: Thanks -- also impressive to be faster than the linter :runner: :rocket:  :)

**Pull request merged**